tools/webdriver/webdriver/bidi/modules/bluetooth.py

```diff
@@ -1,4 +1,4 @@
-from typing import Any, Mapping
+from typing import Any, List, Mapping
 
```

Comment:
Please add wdspec tests for the new commands: https://web-platform-tests.org/writing-tests/wdspec.html#webdriver-bidi

Reply:
Done.

Reply:
For testing `simulate_preconnected_peripheral` or `handle_reuqest_device_prompt_updated`, it would require to simulate a click and then trigger `navigator.bluetooth.requestDevice(...)` like how we do in [infrastructure/testdriver/bidi/bluetooth/simulate_preconnected_peripheral.https.html](https://github.com/web-platform-tests/wpt/blob/a40398d954fdb84328ed2a7e0aa15e863a651fcc/infrastructure/testdriver/bidi/bluetooth/simulate_preconnected_peripheral.https.html). I wonder if wdspec test provides any framework for doing things like appending a button to a page and then simulating a click?

Reply:
> For testing `simulate_preconnected_peripheral` or `handle_reuqest_device_prompt_updated`, it would require to simulate a click and then trigger `navigator.bluetooth.requestDevice(...)` like how we do in [infrastructure/testdriver/bidi/bluetooth/simulate_preconnected_peripheral.https.html](https://github.com/web-platform-tests/wpt/blob/a40398d954fdb84328ed2a7e0aa15e863a651fcc/infrastructure/testdriver/bidi/bluetooth/simulate_preconnected_peripheral.https.html). I wonder if wdspec test provides any framework for doing things like appending a button to a page and then simulating a click?

You can evaluate `document.body.click()` with `user_activation: True`. It will be considered as a user action.

Reply:
I prepared a follow-up PR extending wdspec tests: https://github.com/web-platform-tests/wpt/pull/50796